back/app/controllers/web_api/v1/insights/phase_insights_controller.rb

```diff
@@ -25,6 +38,29 @@ def set_phase
           @phase = Phase.find params[:phase_id]
           authorize @phase
         end
+
+        def validate_and_find_custom_field(group_by)
+          return nil if group_by.blank?
+
+          custom_field = CustomField.find_by(key: group_by)
+
+          unless custom_field
+            render json: { errors: { group_by: [{ error: 'custom_field not found with the key provided' }] } }, status: :unprocessable_entity
+            return nil
+          end
+
+          unless custom_field.resource_type == 'User'
+            render json: { errors: { group_by: [{ error: 'Invalid custom_field resource_type for grouping' }] } }, status: :unprocessable_entity
+            return nil
+          end
+
+          unless custom_field.support_reference_distribution? || custom_field.key == 'birthyear'
+            render json: { errors: { group_by: [{ error: 'Custom field input_type or key not supported for grouping' }] } }, status: :unprocessable_entity
+            return nil
+          end
+
+          custom_field
+        end
```

Comment:
I should probably tidy this up, so an error variable is either set or not, and if set then one render block renders it.